### PR TITLE
add StorageAccess to specify how to access storages

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
Goal is to stop mmapping append vecs.

#### Summary of Changes
Add `StorageAccess` and pass it through a bunch of places so we can specify how storages should be opened.

Fixes #

### CODE REVIEW

File: accounts-db/src/accounts_file.rs [hunk 1]
```diff
@@ -59,13 +59,24 @@ pub enum AccountsFile {
     TieredStorage(TieredStorage),
 }
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum StorageAccess {
+    #[default]
+    /// storages should be accessed by Mmap
+    Mmap,
+}
+
```
Comment (brooksprumo) [inline accounts-db/src/accounts_file.rs]: nit: Can this be moved below the `impl AccountsFile` block? Or minimally not between the `enum` and the `impl` blocks?

File: accounts-db/src/append_vec.rs [hunk 2]
```diff
@@ -397,7 +405,7 @@ impl AppendVec {
         })
     }
 
-    fn sanitize_layout_and_length(&self) -> (bool, usize) {
+    pub fn sanitize_layout_and_length(&self) -> (bool, usize) {
```
Comment (brooksprumo) [inline accounts-db/src/append_vec.rs]: Why `pub`?

File: accounts-db/src/accounts_db.rs [hunk 3]
```diff
@@ -6385,7 +6388,8 @@ impl AccountsDb {
 
             // If the above sizing function is correct, just one AppendVec is enough to hold
             // all the data for the slot
-            assert!(self.storage.get_slot_storage_entry(slot).is_some());
+            let storage = self.storage.get_slot_storage_entry(slot);
+            assert!(storage.is_some());
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: I assume `storage` will be used in a later PR?

File: runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs [hunk 4]
```diff
@@ -67,6 +68,8 @@ pub(crate) struct SnapshotStorageRebuilder {
     num_collisions: AtomicUsize,
     /// Rebuild from the snapshot files or archives
     snapshot_from: SnapshotFrom,
+    /// true if all append vecs must be mmapped
+    storage_access: StorageAccess,
```
Comment (brooksprumo) [inline runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs]: IMO this comment could be reworded, as it makes me expect the `storage_access` is a `bool`.

**Review by HaoranYi — DISMISSED** [review]
No functional changes! lgtm.

File: accounts-db/src/accounts_file.rs (hunk 1 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_file.rs]: moved

File: accounts-db/src/append_vec.rs (hunk 2 above)
Comment (jeffwashington) [reply accounts-db/src/append_vec.rs]: removed

File: runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs (hunk 4 above)
Comment (jeffwashington) [reply runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs]: done

File: accounts-db/src/accounts_db.rs (hunk 3 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs]: removed

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Pull request merged**